### ISSUE TITLE
Intro Question about model.

### ISSUE THREAD

**Issue by BoPengGit** [namisan/mt-dnn#26]
Hi,

Thanks for the repo and your model. Thanks for your time. 

So I have an intro question about the model. 

So this model takes a pretrained BERT model and then jointly fine tunes it on 4 tasks: single sentence classification, pairwise text similarity, pairwise text classification and pairwise ranking.
For pairwise text classification it has an additional stochastic answer network attached to the head but for the other three tasks it just attaches the specific loss function for the three tasks.

The whole network is trained end to end.

Is this correct?

Thanks for the help and God bless!

**Comment by namisan** [comment]
Yes, it is. The whole network is trained e2e.
